### ISSUE TITLE
Login always failed after given authorization

### ISSUE THREAD

**Issue by Giobar97** [playgameservices/play-games-plugin-for-unity#2650]
Hi, I'm having a lot of problems with this plugin.
I spent last 4 days trying to make it work, and seems like it will never end giving me troubles.

I managed to solve all the previous problems, and now I am at the point where finally the login popup shows up (That one asking you to confirm you're giving the authorization for blahblahblah). 
But after I authorize it, the authorization method returns false anyway, making the login fail.

I've already tried setting in my API Credentials SHA1 from Upload Certificate and then with App Signin, as I've read around the web that it might just be the problem.
Both of them results in the same failing login.

I've checked AndroidManifest too, as I've read that there should be "\u003" before the APP_ID number, but it's already auto-generated the right way, so this is not the problem.

I assured to put my email in the whitelist of emails who can test services on my app, as my app is still in alpha testing and not published. I tried with different emails too and with my friends' (whose emails are whitelisted) smartphones too.

I checked the package name set in API Credentials, and it's correct, same for the Client ID you have to put in Google Play Service Setup from Unity's Editor, before you can start to use this API, that's correct too.

I tried to delete the Client to create another one and repeated all the checks I wrote up here, still nothing.

All the posts I've read about this problem just tell to do what I wrote, so I totally don't know what to do now to make it work.

I always downloaded the app from the Play Store to make these checks.

```
    using System.Collections;
    using System.Collections.Generic;
    using UnityEngine;
    using GooglePlayGames;
    using GooglePlayGames.BasicApi;
    using UnityEngine.SocialPlatforms;
    using UnityEngine.SceneManagement;
    
    
        public class GoogleLoginInitialization : MonoBehaviour
        {
        
            [SerializeField] GameObject italianPopup;
            [SerializeField] GameObject englishPopup;
        
            GameObject loginPopup;
        
            void Start()
            {
                SetPopupLanguage();
                InitializeLogin();
                TryLogin();
            }
        
            private void SetPopupLanguage()
            {
                if (Application.systemLanguage == SystemLanguage.Italian)
                {
                    loginPopup = italianPopup;
                }
                else
                {
                    loginPopup = englishPopup;
                }
            }
        
            private void InitializeLogin()
            {
                PlayGamesClientConfiguration config = new PlayGamesClientConfiguration.Builder()
                    .EnableSavedGames()
                    .RequestEmail()
                    .RequestIdToken()
                    .Build();
                PlayGamesPlatform.InitializeInstance(config);
                PlayGamesPlatform.Activate();
            }
        
            private void TryLogin()
            {
                Social.localUser.Authenticate((bool success) => { CheckAuthenticationSuccess(success); });
            }
        
            private void CheckAuthenticationSuccess(bool success)
            {
                if(success == true)
                {
                    SceneManager.LoadScene("Start Menu");
                }
                else
                {
                    loginPopup.SetActive(true);
                }
            }
        
            private void LoginButtonPressed()
            {
                loginPopup.SetActive(false);
                TryLogin();
            }
        
            private void OkButtonPressed()
            {
                Application.Quit();
            }
        }
```

This is the script, it's the only code running in the scene and it's the first scene loading, so I'm sure there's no other code intefering with it.

Does someone have a solution for this mess?

**Comment by legas1** [comment]
Hi, can you show complete logcat log?

**Comment by Giobar97** [comment]
Could you explain me how can I get this log?

**Comment by legas1** [comment]
Sure, I presume you have Android SDK installed which contains adb.exe and that adb is specified in system/user PATH (so you can use it directly in command line like "adb whatever"). Then you connect Android device with USB Debugging enabled (in developer settings) and if you have proper drivers installed, device will be shown in "adb devices" command. Lastly you gather the log, firstly clean log via "adb logcat --clear", secondly run application and then capture log via "adb logcat > someFile.txt". Fo further information about adb, see this link https://developer.android.com/studio/command-line/adb or other sources.

**Comment by Giobar97** [comment]
I presume I should enable the debugging mode in PlayGames platform too, isn't it?
However, I'll post the log in about 20 minutes, the time for me to come back home.

Thank you very much for your help

**Comment by Giobar97** [comment]
Ok, I captured the log (seems that I didn't need to enable the debugging mode in PlayGamesPlatform)
and I found these two blocks of errors in the log: https://pastebin.com/rFZeuvhS

I don't understand why, as my SHA1 is correct (And I tried both SHA1), and the App ID is correct too!

![screen](https://user-images.githubusercontent.com/53182569/61866285-31e6ff00-aed5-11e9-8633-5262a7fbd164.png)
This is the Client screen, and things are set up as in the log.

![Annotazione 2019-07-25 120957](https://user-images.githubusercontent.com/53182569/61866346-4f1bcd80-aed5-11e9-8ad4-1ecc5907103a.png)
This is my Credential page in the API Console, and other than the auto-generated Web Client there's nothing else that could interfere with my client.

![Annotazione 2019-07-25 120846](https://user-images.githubusercontent.com/53182569/61866557-a7eb6600-aed5-11e9-8e66-851460c45180.png)
This is how I set up the plugin in Unity.

![Annotazione 2019-07-25 121452](https://user-images.githubusercontent.com/53182569/61866650-d5d0aa80-aed5-11e9-8d53-53b676d0a049.png)
This is my Linked Apps screen in Services Console.

![InkedAnnotazione 2019-07-25 121558](https://user-images.githubusercontent.com/53182569/61866775-1cbea000-aed6-11e9-801b-0ed90aba6d7d.jpg)
And this is my Test screen from Services Console, and that one is the email which I use to test my app on my smartphone.

**Comment by legas1** [comment]
As far I can see, its related to your configuration, few things that come to my mind:
1. If you use Gradle template and debug build (in release its fine) you are not including sign config in debug builds (as its default configured to have debug sign from Unity).
2. OAuth consent is not filled in Api Console.
3. Credentials config (not sure, from images).

**Comment by Giobar97** [comment]
![Annotazione 2019-07-25 134253](https://user-images.githubusercontent.com/53182569/61871933-5eedde80-aee2-11e9-8db9-445b9f093f6d.png)
![Annotazione 2019-07-25 134317](https://user-images.githubusercontent.com/53182569/61871941-63b29280-aee2-11e9-99b3-7a2fd18e6ba2.png)
These are my project settings.

![Annotazione 2019-07-25 134349](https://user-images.githubusercontent.com/53182569/61871957-6f05be00-aee2-11e9-9b1d-073b3267dfc5.png)
This is how I build my project.
Could the problem be that "Minify" set to None? I set that to None because I've read somewhere that setting to Pooguard could have been a problem for the aar files to be implemented in the build.
Should I try to build with Proguard?


![Annotazione 2019-07-25 133934](https://user-images.githubusercontent.com/53182569/61872131-d9b6f980-aee2-11e9-9dbb-4d90c25e6a8e.png)
![Annotazione 2019-07-25 133954](https://user-images.githubusercontent.com/53182569/61872133-dae82680-aee2-11e9-90e2-e648f2ed01d2.png)
This is instead my Oauth Consent Screen from API Console. It hasn't been verified yet, but have it to be verified to work even in test mode? Do I really have to wait up to a week to test my app?

**Comment by legas1** [comment]
No, Minify set to None is fine and OAuth Consent looks fine, it does not need to be verified. Can you try to build standalone apk without "Build App Bundle" option check (just test, not sure if it will work) and use keytool.exe or jarsigner.exe from JDK to verify signature of APK/bundle?

**Comment by Giobar97** [comment]
Same problem for apk...

**Comment by legas1** [comment]
Can you please check the signatures via keytool or jarsigner if they match your keystore? Also disable signing from Google in Play Console (at least for now).

Probably not, it should work without Firebase.

**Comment by Giobar97** [comment]
Yes, the signatures are exactly the same

**Comment by muhammadadnan0075** [comment]
Hi, I'm having exactly the same problem and still no success. Already tried everything above but I'm receiving "Authentication canceled" in Debug Log. A month ago it was working fine when i initially implemented it but last week i tried to test it and only the Green pop up window appears for few seconds and disappears after that.

**Comment by muhammadadnan0075** [comment]
Okay so my problem was solved after using [this](https://github.com/playgameservices/play-games-plugin-for-unity/tree/android-java-client/current-build) version of the plugin. It seems it was conflicting with Admob plugin and Play services resolver which now uses AndoidX.

**Comment by Giobar97** [comment]
> Okay so my problem was solved after using this version of the plugin. It seems it was conflicting with Admob plugin and Play services resolver which now uses AndoidX.

Thank you.
Actually I started coding my own login/registering/online-saving system, since no one knew how to solve my problem, and the original developer of the plugin seems like a ghost.
So I decided a few days ago to totally abandon this crappy plugin and finally today I completed my own system.
So thank you very much for posting here the solution to my problem, at least you seemed to have interest in this problem, but I won't go back
